src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -42,6 +42,46 @@
  * &lt;module name=&quot;FinalClass&quot;/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * final class MyClass {  // OK
+ *   private MyClass() { }
+ * }
+ *
+ * class MyClass { // violation, class should be declared final
+ *   private MyClass() { }
+ * }
+ *
+ * class MyClass {
+ *   int field1;
+ *   String field2;
+ *   private MyClass(int value) {
+ *     this.field1 = value;
+ *     field2 = " ";
+ *   }
+ *   public MyClass(String value) { // OK, since it has a public constructor
+ *     this.field2 = value;
+ *     field1 = 0;
```

Comment:
Please add `this` here and and in previous ctor

Reply:
done